docs/features/software-catalog/descriptor-format.md

```diff
@@ -0,0 +1,295 @@
+in total.
+
+The `backstage.io/` prefix is reserved for use by Backstage core components. Some keys such as
+`system` also have predefined semantics.
+
+Values are strings that follow the same restrictions as `name` above.
+
+### `annotations` [optional]
+
+An object with arbitrary non-identifying metadata attached to the entity,
+identical in use to [Kubernetes object annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/).
+
+Their purpose is mainly, but not limited, to reference into external systems. This could
+for example be a reference to the git ref the entity was ingested from, to monitoring
+and logging systems, to pagerduty schedules, etc. Users may add these to descriptor
+YAML files, but in addition to this automated systems may also add annotations, either
+during ingestion into the catalog, or at a later time.
+
+Both the key and the value are strings, subject to the following restrictions.
+
+Keys have an optional prefix followed by a slash, and then the name part which is required.
+The prefix must be a valid lowercase domain name if specified, at most 253 characters in total.
+The name part must be sequences of `[a-zA-Z0-9]` separated by any of `[-_.]`, at most 63 characters
+in total.
+
+The `backstage.io/` prefix is reserved for use by Backstage core components.
+
+Values can be of any length, but are limited to being strings.
+
+## Kind: Component
+
+Describes the following entity kind:
+
+| Field        | Value                  |
+| ------------ | ---------------------- |
+| `apiVersion` | `backstage.io/v1beta1` |
+| `kind`       | `Component`            |
+
+A Component describes a software component. It is typically intimately linked to the source code
+that constitutes the component, and should be what a developer may regard a "unit of software",
+usually with a distinct deployable or linkable artifact.
+
+Descriptor files for this kind may look as follows.
+
+```yaml
+apiVersion: backstage.io/v1beta1
+kind: Component
+metadata:
+  name: artist-web
+  description: The place to be, for great artists
+spec:
+  type: website
+  lifecycle: production
+  owner: artist-relations@example.com
+```
+
+In addition to the [common envelope metadata](#common-to-all-kinds-the-metadata) shape, this kind has the
+following structure.
+
+### `apiVersion` and `kind` [required]
+
+Exactly equal to `backstage.io/v1beta1` and `Component`, respectively.
+
+### `spec.type` [required]
+
+The type of component as a string, e.g. `website`. This field is required.
+
+The software catalog accepts any type value, but an organisation should take great
+care to establish a proper taxonomy for these. Tools including Backstage itself may
+read this field and behave differently depending on its value. For example, a website
+type component may present tooling in the Backstage interface that is specific to just
+websites.
+
+The current set of well-known and common values for this field is:
```

Comment:
We were also thinking about adding this, our values are (copy/paste from our docs): 
- web_frontend - Indicates that the Project is a Web Application’s Frontend Service
- web_backend - Indicates that the Project is a Web Application’s Backend Service
- web_fullstack - Indicates that the Project contains a Web Application’s Full Stack of Services including both Frontend & Backend Services
- mobile - Indicates that the Project is a Mobile Application Frontend App

`web_backend` == `service` but we did this for consistency with the other values.
Consider all this, hopefully you find it is a good idea 👯

Reply:
In general when we go around our company to talk to users when trying to establish a new `kind` or `type`, we find that it's not easy to find values that everybody agrees on. Particularly since the inevitable question becomes, WHY do you want to discern type A from type B? Who consumes the value? What is the actual effect/outcome of switching from one value to the other?

We are pretty coarsely granular in our choices. We tend to choose types that are tied to how humans want to consume the thing and what broad family of behaviors they represent. So we have `service`, which is any kind of deployable service-like thing that is communicated with via APIs, without splitting it into subtypes. The reason is, that the greatest consumer of the field is Backstage itself - it may choose to enable service-specific tooling and subviews when looking at that entity's page. The same tools may not appear when looking at a library type entity, because it isn't ever conceivably deployed to kubernetes for example.

You may want to have finer granularity and that could be fine too. But if you find that they actually BEHAVE the exact same, for example in terms of infra tooling that applies to them, then maybe their difference is better explained in the README of the component or in the description field, than as specific types.

Your web_fullstack service may even be represented as linked to three APIs - one that's serving the frontend, one that's an internal management interface, and one that's the graphql backend part. Maybe a Backstage user can then see at a glance what it does anyway, if the entity view is well made, without subtyping.

Again, your choices, and everybody's needs won't be the same as ours. Therefore it's left open to extension :)

Reply:
It is hard to tell if applies without seeing it in more details.
I'm concerned about if this field MUST have these values or not.
It sounds like backstage will know `service` for instance. If so, then these need to be pretty well defined and making sure covers lots of usecases. Website for instance not sure what is that, maybe anything rendering html. 
Or alternatively Backstage does not know about this, it will render them as it finds them in the DB and will match plugins when those target them but really do not know more than that.